sdk/servicebus/service-bus/src/serializers/queueResourceSerializer.ts

```diff
@@ -0,0 +1,382 @@
+
+      enablePartitioning: getBooleanOrUndefined(rawQueue[Constants.ENABLE_PARTITIONING]),
+      requiresSession: getBooleanOrUndefined(rawQueue[Constants.REQUIRES_SESSION]),
+      enableBatchedOperations: getBooleanOrUndefined(rawQueue[Constants.ENABLE_BATCHED_OPERATIONS]),
+
+      defaultMessageTimeToLive: rawQueue[Constants.DEFAULT_MESSAGE_TIME_TO_LIVE],
+      autoDeleteOnIdle: rawQueue[Constants.AUTO_DELETE_ON_IDLE],
+
+      requiresDuplicateDetection: getBooleanOrUndefined(
+        rawQueue[Constants.REQUIRES_DUPLICATE_DETECTION]
+      ),
+      duplicateDetectionHistoryTimeWindow:
+        rawQueue[Constants.DUPLICATE_DETECTION_HISTORY_TIME_WINDOW],
+      deadLetteringOnMessageExpiration: getBooleanOrUndefined(
+        rawQueue[Constants.DEAD_LETTERING_ON_MESSAGE_EXPIRATION]
+      ),
+      forwardDeadLetteredMessagesTo: rawQueue[Constants.FORWARD_DEADLETTERED_MESSAGES_TO],
+
+      messageCountDetails: getCountDetailsOrUndefined(rawQueue[Constants.COUNT_DETAILS]),
+      supportOrdering: getBooleanOrUndefined(rawQueue[Constants.SUPPORT_ORDERING]),
+      enableExpress: getBooleanOrUndefined(rawQueue[Constants.ENABLE_EXPRESS]),
+
+      authorizationRules: getAuthorizationRulesOrUndefined(rawQueue[Constants.AUTHORIZATION_RULES]),
+      isAnonymousAccessible: getBooleanOrUndefined(rawQueue[Constants.IS_ANONYMOUS_ACCESSIBLE]),
+
+      entityAvailabilityStatus: rawQueue[Constants.ENTITY_AVAILABILITY_STATUS],
+      status: rawQueue[Constants.STATUS],
+      createdAt: rawQueue[Constants.CREATED_AT],
+      updatedAt: rawQueue[Constants.UPDATED_AT],
+      accessedAt: rawQueue[Constants.ACCESSED_AT]
+    };
+    return result;
+  }
+}
+
+/**
+ * Represents settable options on a queue
+ */
+export interface QueueOptions {
+  /**
+   * Determines the amount of time in seconds in which a message should be locked for processing by a receiver. After this period, the message is unlocked and available for consumption by the next receiver. Settable only at queue creation time.
+   */
+  lockDuration?: string;
+
+  /**
+   * The entity's size in bytes.
+   *
+   */
+  sizeInBytes?: number;
+
+  /**
+   * Specifies the maximum queue size in megabytes. Any attempt to enqueue a message that will cause the queue to exceed this value will fail.
+   */
+  maxSizeInMegabytes?: number;
+
+  /**
+   * The entity's message count.
+   *
+   */
+  messageCount?: number;
+
+  /**
+   * Depending on whether DeadLettering is enabled, a message is automatically moved to the DeadLetterQueue or deleted if it has been stored in the queue for longer than the specified time. This value is overwritten by a TTL specified on the message if and only if the message TTL is smaller than the TTL set on the queue. This value is immutable after the Queue has been created.
+   */
+  defaultMessageTimeToLive?: string;
+
+  /**
+   * Specifies the time span during which the Service Bus detects message duplication.
+   */
+  duplicateDetectionHistoryTimeWindow?: string;
```

Comment:
See test-cases for reference, seems to be of type "PT1M", "PT2D" etc.
Given that the values/type fine-tuning to this extent are not supported for other SDKs, this can be added as an improvement later as well

Reply:
Logged #5858 to add better documentation around such options/properties